File: src/main/java/me/rasztabiga/fridgy/products/service/mapper/ProductUnitMapper.java
package me.rasztabiga.fridgy.products.service.mapper;

import java.util.Set;
import me.rasztabiga.fridgy.products.domain.*;
import me.rasztabiga.fridgy.products.service.dto.ProductUnitDTO;
import org.mapstruct.*;

/**
 * Mapper for the entity {@link ProductUnit} and its DTO {@link ProductUnitDTO}.
 */
@Mapper(componentModel = "spring", uses = {})
public interface ProductUnitMapper extends EntityMapper<ProductUnitDTO, ProductUnit> {
    @Named("id")
    @BeanMapping(ignoreByDefault = true)
    @Mapping(target = "id", source = "id")
    ProductUnitDTO toDtoId(ProductUnit productUnit);

    @Named("idSet")
    @BeanMapping(ignoreByDefault = true)
    @Mapping(target = "id", source = "id")
    Set<ProductUnitDTO> toDtoIdSet(Set<ProductUnit> productUnit);
}

File: src/main/kotlin/me/rasztabiga/fridgy/products/service/mapper/ProductUnitMapper.kt
package me.rasztabiga.fridgy.products.service.mapper

import me.rasztabiga.fridgy.products.domain.ProductUnit
import me.rasztabiga.fridgy.products.service.dto.ProductUnitDTO
import org.mapstruct.*

/**
 * Mapper for the entity [ProductUnit] and its DTO [ProductUnitDTO].
 */
@Mapper(componentModel = "spring", uses = [])
interface ProductUnitMapper :
    EntityMapper<ProductUnitDTO, ProductUnit> {

    @Named("id")
    @BeanMapping(ignoreByDefault = true)
    @Mappings(
        Mapping(target = "id", source = "id")
    )
    fun toDtoId(productUnit: ProductUnit): ProductUnitDTO

    @Named("idSet")
    @BeanMapping(ignoreByDefault = true)
    @Mappings(
        Mapping(target = "id", source = "id")
    )
    fun toDtoIdSet(productUnit: Set<ProductUnit>): Set<ProductUnitDTO>
}
